puzzle.py

```diff
@@ -167,17 +175,25 @@ def crossover(self, removed_tils):
     random.shuffle(list_center)
     random.shuffle(list_border)
     random.shuffle(list_corner)
-    # Random rotation on center pils
-    for ind in list_center:
-      ind.rotates(random.randint(0, 4))
+    # Put individuals
     for i, ind in enumerate(self.population):
       if ind is None:
         if i in config.corner_pos:
           self.population[i] = list_corner.pop()
         elif i in config.border_pos:
           self.population[i] = list_border.pop()
         else:
-          self.population[i] = list_center.pop(random.randrange(len(list_center)))
+          self.population[i] = list_center.pop()
+          # Check for best rotation
+          rotation = 0
+          fitness = eval.eval_individual_score(self.population, i)
+          for r in (range(1, 4)):
+            self.population[i].rotate()
+            if eval.eval_individual_score(self.population, i) > fitness:

```

Comment:
For now the eval is only on the individual fitness and I thougth to use a mask, but I think it's better like this because we might use eval on group fitness too, for the next implementation.
I think it's more dynamic that way and it will be easier to implement.